helix-view/src/editor.rs

```diff
@@ -160,6 +160,7 @@ pub struct Editor {
     pub count: Option<std::num::NonZeroUsize>,
     pub selected_register: Option<char>,
     pub registers: Registers,
+    pub macro_recording: Option<(char, Vec<crate::input::KeyEvent>)>,
```

Comment:
The type should be imported

Reply:
I'm also not sure if this should be handled here or in `ui::Editor`, the `.` is implemented there: https://github.com/helix-editor/helix/blob/d08bdfa838098769afc59146b62f9d613d4a7ff4/helix-term/src/ui/editor.rs#L712-L719

Reply:
I did initially place the logic in `ui::editor`, but the issue with that is that it will only track the keypresses that go to normal mode commands, we need macros to track *all* keypresses, so that you can record things like `:` command invocations.

Reply:
Wait a minute, this worked fine before I rebased, but now, it's stopped recording command mode keypresses…

Reply:
Hm… I just tried checking out the pre-rebase commits and it still didn't work, but I could have sworn it had worked before, I guess not?

Reply:
Oh, it looks like the keypress is getting consumed by the prompt layer and never makes it to the `EditorView` to get recorded; I should probably move the key recording up to the compositor.